tests/gold_tests/pluginTest/header_rewrite/rules/rule_set_body_from_remap.conf

```diff
@@ -0,0 +1,24 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+cond %{READ_RESPONSE_HDR_HOOK}
+cond %{CLIENT-URL:PATH} = "remap_success" [OR]
+cond %{CLIENT-URL:PATH} = "200"
+set-body-from http://www.example.com/404.html
+
+cond %{READ_RESPONSE_HDR_HOOK}
+cond %{CLIENT-URL:PATH} = "remap_fail"
+set-body-from http://www.example.com/fail
```

Comment:
So, in all the test cases, we still do the original request to upstream, then entirely throw it away.  Wouldn't more realistic test cases be with conditions checking response status or response headers?

If the intent is to always discard the original response, wouldn't it be better for the operator to execute in the remap pseudo hook, and set an error status there, to block the original request?

Reply:
statichit is an example of a plugin that blocks the request to upstream by setting an error status in the DoRemap funciton:  https://github.com/apache/trafficserver/blob/0d004b3a8f8a63cde8e2e67e9117f2def968a312/plugins/statichit/statichit.cc#L639

Reply:
I have written the tests to replicate what happens in production. Do you have an example of another test that only checks the response status?

Reply:
The status is set from that initial request.